packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -411,9 +435,13 @@ export class FeatureDevController {
         let session
         try {
             session = await this.sessionStorage.getSession(message.tabID)
+            if (session.state.codeGenerationResult?.result !== 'success') {
+                throw new ToolkitError('Invalid state in code generation')
```

Comment:
Is it worth extending a toolkiterror and defining a code just in case these cases are being hit and sending this data to telemetry? I see a couple places like these in the changes that might give us some potential insights if something is going very wrong on the users end

Reply:
Yeah good point! Those pieces of code should normally not be reachable, so it makes sense for us to get special telemetry on them.

I'll extend the ToolkitError for this.